new/collapse-mixlib-install-in-tk.md

```diff
@@ -0,0 +1,171 @@
+
+## Motivation
+```
+As a Kitchen User,
+I want to use simple, clear configuration options,
+so that I can setup Kitchen config files with less complexity and obfuscation.
+```
+
+```
+As a Kitchen Developer,
+I want to maintain a single Mixlib Install code path,
+so that I can benefit from the latest API changes and reduce maintenance risk.
+```
+
+The current Mixlib Install code path used in Kitchen was developed specifically for Kitchen and is tightly coupled with Kitchen's provisioner options. It's referred to as the `ScriptGenerator`. The new API was developed to be used by any project that needs to query Chef product meta-data or generate an installation script. The new API takes into account the specifics of interacting with Chef Software Inc's release infrastructure (packages.chef.io).
+
+## Specification
+This work is separated into 3 phases.
+
+##### Phase 1 - Incrementally add new transparent configuration options
+* All existing configuration options behave as expected
+* Build feature parity with the new options to match existing functionality
+* Setting `product_name` will trigger the new options use cases
+* Maintain both Mixlib Install code paths
+* Document new options in the Kitchen repo
+
+##### Phase 2 - Provide option deprecation warnings and "how-tos"
+* All existing configuration options still behave as expected
+* User is clearly warned when using options that will be deprecated
+* Deprecation warnings include explanations and resolution instructions
+* Inform the community of the upcoming Kitchen changes and reasoning
+
+##### Phase 3 - Make the switch
+* Raise errors along with instructions when using deprecated options 
+* `ScriptGenerator` code path removed from Kitchen
+* docs.chef.io and kitchen.ci documentation updates
+* Inform the community when Kitchen is released
+
+### Phase 1 Details
+Below is a proposal of how the new options `product_version` and `install_strategy` replace `require_chef_omnibus`. It outlines requirements to maintain feature parity with Kitchen's current configuration options. Since `require_chef_omnibus` is a complex option it warranted this level of detail.
+
+#### Provisioner setting require_chef_omnibus use cases
+These are the values for require_chef_omnibus and their intent.
+* `true`: If "chef" is installed don't install anything, otherwise install latest
+* `false`: Don't install anything
+* `latest`: Always install latest version
+* `<version>`: If "chef" is installed don't install anything, otherwise install specifed version. Accepts partial versions.
+
+#### Replacement settings for feature parity
+To improve the user experience and align more closley with the new mixlib-install API I am proposing we replace `require_chef_omnibus` with two new settings: `product_version` and `install_strategy`. `product_version` has already been implemented along with `product_name` and `channel`.
```

Comment:
closley -> closely

I don't know if it's realistic to have a table showing old vs new settings, but I wanted one.